### PR TITLE
Dependency fixes for the Kubernetes resources

### PR DESCRIPTION
**Pull request by bgood**
When using Terraform 0.13.5 the Kubernetes provider was getting initialized too early, which caused the kubernetes resources to fail. This PR add dependencies for the kubernetes resources and the gke_auth module, so that the providers get initialized after the cluster stabilizes.

```
tep #5 - "trigger-gitlab": Step #0 - "create-gitlab": module.gke-gitlab.time_sleep.sleep_for_cluster_fix_helm_6361: Creation complete after 3m0s [id=2020-11-04T00:27:56Z]
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.327Z [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.328Z [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/hashicorp/time/0.6.0/linux_amd64/terraform-provider-time_v0.6.0_x5 pid=772
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.329Z [DEBUG] plugin: plugin exited
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": Error: Post "http://localhost/apis/storage.k8s.io/v1/storageclasses": dial tcp 127.0.0.1:80: connect: connection refused
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":   on .terraform/modules/gke-gitlab/main.tf line 274, in resource "kubernetes_storage_class" "pd-ssd":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":  274: resource "kubernetes_storage_class" "pd-ssd" {
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": Error: Post "http://localhost/api/v1/namespaces/default/secrets": dial tcp 127.0.0.1:80: connect: connection refused
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":   on .terraform/modules/gke-gitlab/main.tf line 286, in resource "kubernetes_secret" "gitlab_pg":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":  286: resource "kubernetes_secret" "gitlab_pg" {
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": Error: Post "http://localhost/api/v1/namespaces/default/secrets": dial tcp 127.0.0.1:80: connect: connection refused
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":   on .terraform/modules/gke-gitlab/main.tf line 296, in resource "kubernetes_secret" "gitlab_rails_storage":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":  296: resource "kubernetes_secret" "gitlab_rails_storage" {
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": Error: Post "http://localhost/api/v1/namespaces/default/secrets": dial tcp 127.0.0.1:80: connect: connection refused
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":   on .terraform/modules/gke-gitlab/main.tf line 311, in resource "kubernetes_secret" "gitlab_registry_storage":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":  311: resource "kubernetes_secret" "gitlab_registry_storage" {
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": Error: Post "http://localhost/api/v1/namespaces/default/secrets": dial tcp 127.0.0.1:80: connect: connection refused
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":   on .terraform/modules/gke-gitlab/main.tf line 329, in resource "kubernetes_secret" "gitlab_gcs_credentials":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":  329: resource "kubernetes_secret" "gitlab_gcs_credentials" {
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab":
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.947Z [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/hashicorp/kubernetes/1.11.4/linux_amd64/terraform-provider-kubernetes_v1.11.4_x4 pid=886
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.947Z [DEBUG] plugin: plugin exited
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.947Z [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/hashicorp/kubernetes/1.11.4/linux_amd64/terraform-provider-kubernetes_v1.11.4_x4 pid=878
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.948Z [DEBUG] plugin: plugin exited
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.948Z [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/hashicorp/template/2.1.2/linux_amd64/terraform-provider-template_v2.1.2_x4 pid=794
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.948Z [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/hashicorp/helm/1.2.4/linux_amd64/terraform-provider-helm_v1.2.4_x4 pid=867
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.948Z [DEBUG] plugin: plugin exited
Step #5 - "trigger-gitlab": Step #0 - "create-gitlab": 2020-11-04T00:27:56.949Z [DEBUG] plugin: plugin exited
```

### CODE REVIEW

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @bgood 
For transitioning to module depends_on and TF 0.13 we will need to 
i) Update versions here and make a breaking change https://github.com/terraform-google-modules/terraform-google-gke-gitlab/blob/5e2f64573419dd31bf246c9fb90ff7f33d194a8e/version.tf#L17
ii) Update our CI images to `0.13` 
https://github.com/terraform-google-modules/terraform-google-gke-gitlab/blob/5e2f64573419dd31bf246c9fb90ff7f33d194a8e/build/int.cloudbuild.yaml#L41

**Comment by bgood** [comment]
Updated the version.tf and bumped the tag for the CI tools.

**Review by bharathkkb — COMMENTED** [review]
You will also have to bump the setup project to `~> 9.0` for TF 0.13 support.
https://github.com/terraform-google-modules/terraform-google-gke-gitlab/blob/5e2f64573419dd31bf246c9fb90ff7f33d194a8e/test/setup/main.tf#L19

File: version.tf [hunk 1]
```diff
@@ -15,36 +15,36 @@
...
     # source is not supported in TF12
     # but is required for TF13
     google = {
-      # source  = "hashicorp/google"
+      source  = "hashicorp/google"
       version = "~> 3.39"
     }
     google-beta = {
-      # source  = "terraform-providers/google-beta"
+      source  = "terraform-providers/google-beta"
```
Comment (bharathkkb) [inline version.tf:27]:
```suggestion
      source  = "hashicorp/google-beta"
```

File: version.tf [hunk 2]
```diff
@@ -15,36 +15,36 @@
  */
 
 terraform {
-  required_version = ">= 0.12.0"
+  required_version = ">= 0.13.0"
   required_providers {
     # source is not supported in TF12
```
Comment (bharathkkb) [inline version.tf:20]: nit: remove comment as we are now using 0.13

**Comment by bharathkkb** [comment]
@morgante any blockers on transitioning this module to 0.13?

**Comment by morgante** [comment]
@bharathkkb It's fine with me.

File: version.tf (hunk 1 above)
Comment (bgood) [reply version.tf]: Done.

File: version.tf (hunk 2 above)
Comment (bgood) [reply version.tf]: Done

**Comment by bharathkkb** [comment]
@bgood Integration test error seems like it is due to https://github.com/terraform-google-modules/terraform-google-gke-gitlab/pull/64#issuecomment-717507395 Is it working locally for you?

**Comment by bgood** [comment]
For the project I'm using the GitLab module in and testing with I have the provider versions set to 3.44.  The GitLab module picks that up because of the ~> and everything is working.

Would you like me to run a test with 3.39 or updates this PR to use 3.44?

Thanks
Ben

**Comment by bgood** [comment]
I also just tested with 3.39.0 and it's working correctly for me.

**Comment by bgood** [comment]
@bharathkkb @morgante

Do you have another logs or data we can look at for these tests that are failing?  I'm failing to reproduce the failures locally using the simple example and my other project.

Thanks
Ben

**Comment by morgante** [comment]
@bgood The logs are in Cloud Build (ping me if you don't have access).

This is the failure:
```
       
       Error: timed out waiting for the condition
       
         on ../../../main.tf line 384, in resource "helm_release" "gitlab":
        384: resource "helm_release" "gitlab" {
```

**Comment by morgante** [comment]
Possibly there's an issue with connectivity from Cloud Build to the cluster?

**Comment by bgood** [comment]
The project I'm using this module in is also deploying GitLab using Cloud Build.  So it could be possible, but there must also be something else different.

Should I make a change to turn on debug level logging for the integration tests and run again?

**Comment by morgante** [comment]
@bgood Yeah give it a try. We can also keep re-running.

**Comment by bgood** [comment]
Looks like the last couple runs succeeded from a deployment perspective, the helm release step only took 7 min.  Unfortunately there was a validation failure. 

With the last run I tried upgrading kitchen to 5.5.0 thinking that might be the problem, but no joy.

**Comment by morgante** [comment]
It looks like the reachability test is getting a 404 error: https://github.com/terraform-google-modules/terraform-google-gke-gitlab/blob/master/test/integration/simple_example/controls/gcloud.rb#L25

Possibly `gitlab_url` is incorrect for the new version?

**Comment by bharathkkb** [comment]
@bgood I just tried running it locally. Helm release works but the test fails and if I try to access the URL via browser I get `default backend - 404`. Have not had a chance to debug yet.

**Comment by bgood** [comment]
Looks like the test are passing now.  The last change I made was to add `gitlab.` to the output string for `gitlab_url`.  That had been removed in a different PR (#62), which also passed tests.

With these runs, it looks to me like `gitlab.` should *be* there for the output value.  Not sure why the test passed on the PR which removed it.

**Review by morgante — APPROVED** [review]

**Pull request merged**